### PR TITLE
Complete documentation on append-only remote repos

### LINKED ISSUE #5497: Running borg init via a borg serve --append-only server will not create an append-only repository.
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

QUESTION

## Describe the problem you're observing.

What exactly does the following mean?

> Running borg init via a borg serve --append-only server will not create an append-only repository.

My understanding is that if the server is configured using the following `authorized_keys`, when initializing a repo using `borg init --encryption=keyfile-blake2 "borg@server:backup"`, `--append-only` on the server is ignored.

```shell
command="borg serve --append-only --restrict-to-repository /home/borg/backup --storage-quota 10G",restrict ssh-rsa ...
```

This doesn't feel logical to me as the server config should supersede client (for security reasons). Also, I would like to use two SSH keys, one with `--append-only` and one without (using a key that requires a YubiKey for increased security).

```shell
command="borg serve --append-only --restrict-to-repository /home/borg/backup --storage-quota 10G",restrict ssh-rsa ...
```

vs

```shell
command="borg serve --restrict-to-repository /home/borg/backup --storage-quota 10G",restrict ssh-rsa ...
```

Can someone please confirm how this works?

### PR DESCRIPTION
**Pull request by Gu1nness**
I have updated both the FAQ and the https://github.com/borgbackup/borg/blob/master/docs/deployment/hosting-repositories.rst files.

Fixes #5497

### CODE REVIEW

File: docs/deployment/hosting-repositories.rst [hunk 1]
```diff
@@ -68,6 +68,15 @@ can be filled to the specified quota.
 If storage quotas are used, ensure that all deployed Borg releases
 support storage quotas.
 
+**Append-only repositorues**
```
Comment (ThomasWaldmann) [inline docs/deployment/hosting-repositories.rst:71]: typo

Comment (ThomasWaldmann) [reply docs/deployment/hosting-repositories.rst:71]: also, compared to the paragraph above, which has some introduction into usage, this one is only pointing out the special stuff.

File: docs/deployment/hosting-repositories.rst [hunk 2]
```diff
@@ -68,6 +68,15 @@ can be filled to the specified quota.
...
+
+Running ``borg init`` via a ``borg serve --append-only`` server will not create
```
Comment (ThomasWaldmann) [inline docs/deployment/hosting-repositories.rst:73]: guess the **not** should be emphasized or bold.

File: docs/faq.rst [hunk 3]
```diff
@@ -183,6 +183,14 @@ This is not a problem anymore.
 
 For more details, see :ref:`checkpoints_parts`.
 
+Do we have to run borg config "backup" append_only 0 and borg config "backup" append_only 1 on the sever to switch append only mode
```
Comment (ThomasWaldmann) [inline docs/faq.rst:186]: "backup"?

did you mean REPO?

also: add a question mark.

Comment (ThomasWaldmann) [reply docs/faq.rst:186]: or rephrase to "How can I switch append-only on and off?"

File: docs/faq.rst [hunk 4]
```diff
@@ -183,6 +183,14 @@ This is not a problem anymore.
...
+-----------------------------------------------------------------------------------------------------------------------------------
+
+You could do that, but using different ssh keys and different entries in
+authorized_keys is much easier and also maybe has less potential of thing going
```
Comment (ThomasWaldmann) [inline docs/faq.rst:190]: add markup so that `authorized_keys` looks like a filename.

File: docs/faq.rst [hunk 5]
```diff
@@ -183,12 +183,12 @@ This is not a problem anymore.
 
 For more details, see :ref:`checkpoints_parts`.
 
-Do we have to run borg config "backup" append_only 0 and borg config "backup" append_only 1 on the sever to switch append only mode
+How can I switch append-only mode on and off?
 -----------------------------------------------------------------------------------------------------------------------------------
 
 You could do that, but using different ssh keys and different entries in
```
Comment (ThomasWaldmann) [inline docs/faq.rst:189]: You could do that (via `borg config REPO append_only 0/1`), but ...

File: docs/faq.rst [hunk 6]
```diff
@@ -183,12 +183,12 @@ This is not a problem anymore.
...
-authorized_keys is much easier and also maybe has less potential of thing going
-wrong somehow.
+``authorized_keys`` is much easier and also maybe has less potential of thing
```
Comment (ThomasWaldmann) [inline docs/faq.rst:190]: things

File: docs/deployment/hosting-repositories.rst (hunk 1 above)
Comment (ThomasWaldmann) [reply docs/deployment/hosting-repositories.rst]: the 2nd item is still todo.

Comment (Gu1nness) [reply docs/deployment/hosting-repositories.rst]: What do you mean exactly ?
The former paragraph is about storage quotas, which seems also a special stuff, no?

Comment (ThomasWaldmann) [reply docs/deployment/hosting-repositories.rst]: the paragraph about quotas first tells general stuff, like how to use them.

this paragraph does not, it just says what does not work like some people could maybe expect.

Comment (ThomasWaldmann) [reply docs/deployment/hosting-repositories.rst]: i'll change the remaining stuff later.

**Pull request merged**